stable/nginx-ingress/templates/controller-cluster-role.yaml

```diff
@@ -0,0 +1,54 @@
+{{- if .Values.controller.rbac.enabled -}}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ template "fullname" . }}
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+      - endpoints
+      - nodes
+      - pods
+      - secrets
+    verbs:
+      - list
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - nodes
```

Comment:
The controller is attempting to request a list of namespaces and due to lack of permissions throws an exception `(get namespaces stage-1-osf)`. This command option essentially allows you to have the controller only watch a specified namespace for ingress' tagged with `nginx`, rather than a single global ingress nginx controller for the entire cluster.

In our case we are evaluating dedicated ip/ingress for each significant component of a larger application, thinking more towards a potential future of adding ModSecurity at this controller ingress point.